src/libraries/Native/Unix/System.Security.Cryptography.Native/pal_evp_cipher.c

```diff
@@ -126,8 +126,26 @@ int32_t CryptoNative_EvpCipherReset(EVP_CIPHER_CTX* ctx)
     //
     // But since we have a different object returned for CreateEncryptor
     // and CreateDecryptor we don't need to worry about that.
+    uint8_t* iv = NULL;
 
-    return EVP_CipherInit_ex(ctx, NULL, NULL, NULL, NULL, KEEP_CURRENT_DIRECTION);
+#ifdef NEED_OPENSSL_3_0
+    // OpenSSL 3.0 alpha 13 does not properly reset the IV. Work around that by
+    // asking for the original IV, and giving it back.
```

Comment:
Is there an issue in openssl's bug database tracking this somewhere?

Reply:
Yeah, linked off of https://github.com/dotnet/runtime/issues/46526#issuecomment-809667862 (didn't see a need for multiple links from our repository to their issue).

Their first fix didn't fix things fully, but they reactivated the issue.